### PR TITLE
ADDON-37153 Bug Fixes

### PR DESCRIPTION
**Pull request by dhruvanshus-crest**
Jira: https://jira.splunk.com/browse/ADDON-37153

Bugs Fixed :
- In the background , it automatically Selects First Tab on click of edit of account
- Placeholders missing where required field is false
- Account deletion not working when account name with special characters is entered from backend

### CODE REVIEW

File: src/main/webapp/components/ControlWrapper.jsx [hunk 1]
```diff
@@ -33,14 +33,22 @@ class ControlWrapper extends React.PureComponent {
         this.controlType = this.isString(props.entity.type)
             ? CONTROL_TYPE_MAP[props.entity.type]
             : null;
+        // Add 'optional' placeholder for optional field
+        this.options = this.props.entity.options;
+        if (this.props.entity.required === false || this.props.entity.required === undefined) {
```
Comment (harshpatel-crest) [inline src/main/webapp/components/ControlWrapper.jsx:38]: Can we get away by just checking `if(!this.props.entity.required)`?

File: src/main/webapp/components/table/CustomTable.jsx [hunk 2]
```diff
@@ -115,7 +115,7 @@ function CustomTable({
             }
         },
         // eslint-disable-next-line react-hooks/exhaustive-deps
-        [entityModal]
+        [entityModal][history.location.search]
```
Comment (harshpatel-crest) [inline src/main/webapp/components/table/CustomTable.jsx:118]:
```suggestion
        [entityModal, history.location.search]
```

**Review by harshpatel-crest — CHANGES_REQUESTED** [review]

File: src/main/webapp/components/ControlWrapper.jsx (hunk 1 above)
Comment (dhruvanshus-crest) [reply src/main/webapp/components/ControlWrapper.jsx]: Done

File: src/main/webapp/components/table/CustomTable.jsx (hunk 2 above)
Comment (dhruvanshus-crest) [reply src/main/webapp/components/table/CustomTable.jsx]: Done

**Review by mchavda-splunk — APPROVED** [review]

**Pull request merged**